waiter/src/waiter/service_description.clj

```diff
@@ -831,10 +832,26 @@
                              (retrieve-token-update-epoch-time token (token->token-parameters token) version)))
                       (reduce utils/nil-safe-max nil))))))
 
+(def ^:const waiter-config-token-path ["env" "WAITER_CONFIG_TOKEN"])
+
 (defn- assoc-token-in-env
   "Attaches the token sequence as an environment variable in the service description."
   [service-description token-sequence]
-  (assoc-in service-description ["env" "WAITER_CONFIG_TOKEN"] (str/join "," token-sequence)))
+  (assoc-in service-description waiter-config-token-path (str/join "," token-sequence)))
+
+(defn- promote-default-to-exclusive?
+  "Determines if a service description should be promoted to exclusive mode based on reference last update time."
+  [reference-update-epoch-time]
+  (let [exclusive-promotion-start-epoch-time (config/retrieve-exclusive-promotion-start-epoch-time)]
+    (< exclusive-promotion-start-epoch-time reference-update-epoch-time)))
+
+(defn adjust-waiter-config-token
+  "Dissociates the exclusive mapping token from the environment variable if it was added as part of default promotion after the promotion start time."
+  [service-description token-service-mapping reference-update-epoch-time]
+  (cond-> service-description
+    (and (= "default" token-service-mapping)
+         (not (promote-default-to-exclusive? reference-update-epoch-time)))
+    (utils/dissoc-in waiter-config-token-path)))
```

Comment:
Is it possible that this will change the service-id? I've seen services where `env: {}`, but I think this would change those service descriptions to delete the `env` entirely.

Reply:
Nice observation, I will remove the `env` if it becomes empty.

Reply:
Done.